### PR TITLE
backend, dist-git: unify run dispatchers

### PR DESCRIPTION
**Pull request by Bob131**
A hurdle when trying to stand-up a simple single-machine Copr deployment is that both the copr-backend and copr-dist-git packages install a script to /usr/bin/copr-run-dispatcher, preventing the user from asking that both be installed on the same system with dnf.

This commit unifies the two scripts into a single script and has it shipped as part of python-copr-common instead.

### CODE REVIEW

**Comment by Bob131** [comment]
Regarding https://github.com/fedora-copr/copr/issues/3007#issuecomment-1818989419:

>One issue after a quick patch review: please don't move the script from `dist-git` to `common`.

Do you mean "don't move the script from `dist-git`" or "don't move the script to `common`"? 
 - If the former: keeping the script in `dist-git` would mean `backend` would have to have a dependency on `dist-git`, right?
 - If the latter: where should it go instead?

**Comment by nikromen** [comment]
> Do you mean "don't move the script from dist-git" or "don't move the script to common"?

I think Pavel meant both - not to move it to `common` package and also keep it in `dist-git` (so we don't have to sync it between common and dist-git/backend).

Quickly looking to the issue - I would suggest renaming the `copr-run-dispatcher` to `sopr-run-dispatcher-(backend/dist-git)` to prevent it from installing to the same file - WDYT of it @praiskup ?

**Comment by nikromen** [comment]
Thank you for the PR!

> Do you mean "don't move the script from dist-git" or "don't move the script to common"?

I think Pavel meant both - not to move it to `common` package and also keep it in `dist-git` (so we don't have to sync it between common and dist-git/backend).

Quickly looking to the issue - I would suggest renaming the `copr-run-dispatcher` to `sopr-run-dispatcher-(backend/dist-git)` to prevent it from installing to the same file - WDYT of it @praiskup ?

File: common/run/copr-run-dispatcher [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+dispatchers from our systemd unit files.
+"""
+
+import sys
+
+
+def _main():
+    request = sys.argv[1]
+    if request == "actions" or request == "builds":
+        from copr_backend.daemons.action_dispatcher import ActionDispatcher
```
[the remark this answers is not in the document]
Comment (Bob131) [reply common/run/copr-run-dispatcher:14]: The reason for guarding the imports behind the request type check is to make it into a sort of 'soft' dependency:

 - Without `copr-backend` and `copr-dist-git` installed, there is no reason anything should be executing `/usr/bin/copr-run-dispatcher`
 - Without `copr-dist-git` installed, there is no reason anything should be running `/usr/bin/copr-run-dispatcher import`
 - etc

It doesn't seem that different to me than the common `try: import ...` idiom.

**Comment by Bob131** [comment]
> Quickly looking to the issue - I would suggest renaming the `copr-run-dispatcher` to `sopr-run-dispatcher-(backend/dist-git)` to prevent it from installing to the same file - WDYT of it @praiskup ?

...that does make more sense!

I guess I was thinking that the decision to have both sets of actions dispatched through a single script was made for a particular reason. But if we can just rename the scripts then I'm happy with that :slightly_smiling_face:

File: common/run/copr-run-dispatcher (hunk 1 above)
[the remark this answers is not in the document]
Comment (nikromen) [reply common/run/copr-run-dispatcher:14]: Well yes, it's possible... but we'd like to preserve no dependencies (even soft) on copr-common between other copr packages

**Comment by praiskup** [comment]
> > Quickly looking to the issue - I would suggest renaming the `copr-run-dispatcher` to `sopr-run-dispatcher-(backend/dist-git)` to prevent it from installing to the same file - WDYT of it @praiskup ?
> 
> ...that does make more sense!
> 
> I guess I was thinking that the decision to have both sets of actions dispatched through a single script was made for a particular reason. But if we can just rename the scripts then I'm happy with that 🙂

Yes, this sounds good to me as well.

**Comment by Bob131** [comment]
I've pushed a patch that renames the scripts, as discussed. The branch name and PR title are now a little misleading, but I didn't think it was worth opening a new PR :wink:

A forewarning that I haven't done a great deal of testing here; I'm hoping that the CI setup will catch anything I've messed up, but if need be I can try building the packages and setting up a test environment later.

**Review by praiskup — APPROVED** [review]
LGTM

**Comment by praiskup** [comment]
Thank you.

**Pull request merged**